## Importación de módulos 

In [1]:
# Librerías dedicadas al procesamiento de los datos
import numpy as np
import pandas as pd

# Librerías para la visualización de los datos
import matplotlib.pyplot as plt
import seaborn as sns

# Librerías personales
from Lectura_datos.Lectura_datos import lectura_inicial_datos

## Lectura de los datos del Bitcoin

Los datos escogidos en esta primera aproximación al problema de la predicción del precio del Bitcoin se encuentran en la siguiente dirección web: **https://www.kaggle.com/mczielinski/bitcoin-historical-data/activity**. El rango temporal de dichos datos comienza el 1 de Enero de 2012 y finaliza el 31 de Diciembre de 2020, con un intervalo de 1 minuto entre ellos. El peso de la base de datos es de 300 MB y contiene más de dos millones de registros. 

Las variables que componen la base de datos a explorar son las siguientes: 
* Tiempo del registro (en modo Unix-timestamp).
* Valor de apertura endolares(USD) del precio en el minuto registrado.
* Valor más alto del precioen el minuto registrado.
* Valor más bajo del precio en el minuto registrado. 
* Valor de cierre del precio en el minuto registrado.
* Volumen en el minutoregistrado del bitcoin.
* Volumen en el minuto registrado de la divisa (USD).
* Precio ponderado en dolares en ese minuto registrado. 

Las filas o registros sin ninguna operación o actividad están instanciadas con valores de tipo *NaN* en sus columnas (exceptuando la fecha). La falta de registros, o presencia de saltos, puede deberse a la ausencia de intercambios, a problemas con su API o a errores técnicos en la presentación o recopilación de datos.

In [2]:
raw_data_bitcoin = lectura_inicial_datos()

## 1. Análisis Exploratorio
El proceso de exploración de los datos se ha realizado para:
1. Comprender los datos disponibles.
2. Determinar la cantidad de valores nulos y sus posibles causas.
3. Encontrar correlaciones entre las variables.


### 1.1 Observaciones preliminares

In [3]:
raw_data_bitcoin.head()

Open  High   Low  Close  Volume_(BTC)  Volume_(Currency)  \
Timestamp                                                                       
2011-12-31 08:52:00  4.39  4.39  4.39   4.39      0.455581                2.0   
2011-12-31 08:53:00   NaN   NaN   NaN    NaN           NaN                NaN   
2011-12-31 08:54:00   NaN   NaN   NaN    NaN           NaN                NaN   
2011-12-31 08:55:00   NaN   NaN   NaN    NaN           NaN                NaN   
2011-12-31 08:56:00   NaN   NaN   NaN    NaN           NaN                NaN   

                     Weighted_Price  
Timestamp                            
2011-12-31 08:52:00            4.39  
2011-12-31 08:53:00             NaN  
2011-12-31 08:54:00             NaN  
2011-12-31 08:55:00             NaN  
2011-12-31 08:56:00             NaN

In [4]:
raw_data_bitcoin.shape

(4727777, 7)

* La base de datos escogida está compuesta por 8 columnas y 4727777 filas. 
* 'Weighted_Price' es la variable objetivo del dataset.
* Las variables 'Timestamp', 'Open', 'High', 'Low', 'Close', 'Volume_(BTC)' y 'Volume_(Currency)' serán utilizadas por los distintos modelos para obtener las predicciones deseadas.

In [5]:
raw_data_bitcoin.info()

<class 'pandas.core.frame.DataFrame'>
DatetimeIndex: 4727777 entries, 2011-12-31 08:52:00 to 2020-12-31 01:00:00
Data columns (total 7 columns):
 #   Column             Dtype  
---  ------             -----  
 0   Open               float64
 1   High               float64
 2   Low                float64
 3   Close              float64
 4   Volume_(BTC)       float64
 5   Volume_(Currency)  float64
 6   Weighted_Price     float64
dtypes: float64(7)
memory usage: 288.6 MB


**Variables numéricas**

In [6]:
# Lista de las variables numéricas del dataset
raw_data_bitcoin.select_dtypes(include=['number']).columns

Index(['Open', 'High', 'Low', 'Close', 'Volume_(BTC)', 'Volume_(Currency)',
       'Weighted_Price'],
      dtype='object')

Características generales de las variables númericas:

In [7]:
raw_data_bitcoin.select_dtypes(include=['number']).describe().round(decimals = 3)

Open         High          Low        Close  Volume_(BTC)  \
count  3484305.000  3484305.000  3484305.000  3484305.000   3484305.000   
mean      4570.658     4573.721     4567.413     4570.624         9.423   
std       4775.105     4778.620     4771.352     4775.094        31.013   
min          3.800        3.800        1.500        1.500         0.000   
25%        432.190      432.430      432.000      432.200         0.392   
50%       3085.010     3093.250     3078.010     3084.410         1.937   
75%       8158.010     8162.850     8152.790     8157.310         7.300   
max      28997.380    29010.320    28963.660    28997.380      5853.852   

       Volume_(Currency)  Weighted_Price  
count       3.484305e+06     3484305.000  
mean        3.315691e+04        4570.625  
std         1.129107e+05        4775.090  
min         0.000000e+00           3.800  
25%         4.147940e+02         432.175  
50%         3.305199e+03        3085.478  
75%         2.158297e+04        8157.964  
max         1.044599e+07       28985.886